src/Accelerators/Accelerator.cpp

```diff
@@ -10,6 +10,8 @@
 //
 // Accelerator base class
 //
+// To enable a new accelerator, add the header include, an extern of the
+// subclass and pushback that subclass variable onto acceleratorTargets.
 //===----------------------------------------------------------------------===//
 #include "src/Accelerators/Accelerator.hpp"
 #include <iostream>
```

Comment:
at line 21 we have:
```
std::vector<Accelerator *> *Accelerator::acceleratorTargets;
```
which is an initialize pointer to a vector.  Suggest to declare a SmallPtrSet (LLVM data type) instead as a private member variable of the `Accelerator class`, that way the constructor need not `new` a new vector:

```
class Accelerator {
public:
  . . .
private:
  /// Accelerators available.
  static llvm::SmallPtrSet<Accelerator *, 2> accelerators;
};
```

The in `Accelerator.cpp`:
```
namespace onnx_mlir {
namespace accel {

llvm::SmallPtrSet<Accelerator *, 2> Accelerator::accelerators;
```

```

Reply:
@etiotto So, basically thats what I have, except that it isn't a SmallPtrSet. What is the advantage of changing to that?

The new in the constructor was an accident of debugging and I will remove that.

Reply:
LLVM's SmallPtrSet etc.. "handles small sets extremely efficiently, but gracefully handles extremely large sets without loss of efficiency." So, generally speaking, these are better than the STL containers and I have seen LLVM STL 'like' containers been used quite a bit in LLVM codebase. Here is more info https://llvm.org/docs/ProgrammersManual.html#dss-smallset